### PR TITLE
[MRG] add plot_convergence

### PR DESCRIPTION
**Pull request by glouppe**
This PR adds a convenience function for drawing one or several convergence traces. Notebooks have been updated (and simplified) accordingly.

### CODE REVIEW

**Comment by betatim** [comment]
download 8
download 7
download 6

The three relevant plots.

File: skopt/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """
 Scikit-Optimize, or `skopt`, is a simple and efficient library
-for sequential model-based optimization, accessible to everybody
```
Comment (betatim) [inline skopt/__init__.py]: (any idea why we keep having whitespace at the end of the lines?)

File: skopt/plots.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    * `args[i]` [`OptimizeResult`, list of `OptimizeResult`, or tuple]:
+        The result(s) for which to plot the convergence trace.
+
+        - if `OptimizeResult`, then draw the corresponding single trace;
+        - if list of `OptimizeResult`, then draw the corresponding convergence
+          traces in light, along with the average convergence trace;
+        - if tuple, then `args[i][0]` should be a string label and `args[i][1]`
+          an `OptimizeResult` or a list of `OptimizeResult`.
+    """
+    colors = cm.viridis(np.linspace(0.25, 1.0, len(args)))
```
Comment (betatim) [inline skopt/plots.py]: Nice! Didn't know this

**Comment by betatim** [comment]
What do other libraries that create plots do in terms  of using `plt.plot` vs `ax.plot`? My intuition would be that we should create a new figure, plot on it, and return it. Will check `seaborn` and `pandas` to see what they do

**Comment by glouppe** [comment]
> What do other libraries that create plots do in terms of using plt.plot vs ax.plot? My intuition would be that we should create a new figure, plot on it, and return it. Will check seaborn and pandas to see what they do

I could call `fig = plt.figure()` at the beginning and then return it without calling `plt.show()`? (but then it would be a bit less convenient to use)

**Comment by betatim** [comment]
Seems like a lot of `seaborn` and `pandas` plotting stuff does:

```python
def plot_me(..., ax=None):
  if ax is None:
    ax = plt.gca()
  ...
  return ax
```

I'd just copy what they are doing in the absence of knowledge/experience with this subject.

**Comment by betatim** [comment]
Legacy python 🐍 says hello via travis: `def f(*args, kw1=1, kw2=2)` is verboten :-/

File: skopt/__init__.py (hunk 1 above)
Comment (glouppe) [inline skopt/__init__.py]: Nope :s

**Comment by glouppe** [comment]
Thanks for the review, comments addressed!

**Comment by betatim** [comment]
🍦🍦🍦

**Comment by MechCoder** [comment]
Thanks! I will be using this today to compare the performance of the acquisition minimization techniques

**Pull request merged**